modules/xfem/src/constraints/XFEMCohesiveConstraint.C

```diff
@@ -0,0 +1,235 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "XFEMCohesiveConstraint.h"
+
+// MOOSE includes
+#include "Assembly.h"
+#include "ElementPairInfo.h"
+#include "FEProblem.h"
+
+registerMooseObject("XFEMApp", XFEMCohesiveConstraint);
+
+InputParameters
+XFEMCohesiveConstraint::validParams()
+{
+  InputParameters params = XFEMMaterialManagerConstraint::validParams();
+  params.addClassDescription("Enforce a cohesive zone traction-separation-law.");
+  params.addParam<Real>("stiffness", 9.6628, "Initial stiffness.");
```

Comment:
'stiffnes', 'max_traction', and 'Gc' should all be required parameters rather than having these (unusual) defaults.

Reply:
Yeah, I don't know that the deal with those is :-D. I'll clean it up.